cargo-concordium/src/schema_json.rs

```diff
@@ -270,6 +270,31 @@ pub fn write_bytes_from_json_schema_type<W: Write>(
                 bail!("JSON Object with one field required for an Enum")
             }
         }
+        Type::EnumEvent(variants_ty) => {
+            if let Value::Object(map) = json {
+                ensure!(map.len() == 1, "Only one variant allowed");
+                let (variant_name, fields_value) = map.iter().next().unwrap(); // Safe since we already checked the length
+                let schema_fields_opt = variants_ty
+                    .iter()
+                    .enumerate()
+                    .find(|(_, (variant_name_schema, _, _))| variant_name_schema == variant_name);
+                if let Some((i, (_, variant_fields, tag))) = schema_fields_opt {
+                    if variants_ty.len() <= 256 {
+                        out.write_u8(i as u8)
+                            .map_err(|_| anyhow!("Failed writing"))?;
+                    } else {
+                        bail!("EnumEvents with more than 256 variants are not supported.");
+                    };
+                    write_bytes_from_json_schema_fields(variant_fields, fields_value, out)?;
+                    out.write_u8(*tag).map_err(|_| anyhow!("Failed writing"))
```

Comment:
Shouldn't this be
```suggestion
                    if variants_ty.len() <= 256 {
                        out.write_u8(*tag)
                            .map_err(|_| anyhow!("Failed writing"))?;
                    } else {
                        bail!("EnumEvents with more than 256 variants are not supported.");
                    };
                    write_bytes_from_json_schema_fields(variant_fields, fields_value, out)?;
```
and you don't even need the `enumerate`?